tests/components/modbus/conftest.py

```diff
@@ -77,14 +94,12 @@ async def mock_test_state(hass, request):
     return request.param
 
 
-# dataclass
-class ReadResult:
-    """Storage class for register read results."""
-
-    def __init__(self, register_words):
-        """Init."""
-        self.registers = register_words
-        self.bits = register_words
+@pytest.fixture
+async def mock_ha(hass):
+    """Load homeassistant to allow service calls."""
+    assert await async_setup_component(hass, "homeassistant", {})
+    await hass.async_block_till_done()
+    yield mock_modbus
```

Comment:
Why are we yielding `mock_modbus`? Doesn't look like we need to yield anything.

Reply:
You are absolutely right, I had the impression I needed to yield the mock in order to have access to it in the test method, but it works without.

Done.

Reply:
`mock_modbus` is another fixture which does yield a mock.